service/history/shardContext.go

```diff
@@ -722,23 +722,32 @@ func (s *shardContextImpl) emitShardInfoMetricsLogsLocked() {
 	}
 	diffTimerLevel := maxTimerLevel.Sub(minTimerLevel)
 
-	if logWarnTransferLevelDiff < diffTransferLevel || logWarnTimerLevelDiff < diffTimerLevel {
+	replicationLag := s.transferMaxReadLevel - s.shardInfo.ReplicationAckLevel
+	transferLag := s.transferMaxReadLevel - s.shardInfo.TransferAckLevel
+	timerLag := time.Since(s.shardInfo.TimerAckLevel)
+
+	if logWarnTransferLevelDiff < diffTransferLevel ||
+		logWarnTimerLevelDiff < diffTimerLevel ||
+		logWarnTransferLevelDiff < replicationLag ||
```

Comment:
Should we have a separate threshold for replicationLag?

Reply:
if there is no replication generated on this shard, then the replicationLag will increase since the replication ack level is unchanged.

emitting metrics here helps us understand the shard status, but we should not use this as an indicator whether to alarm.

i am dropping emit warn log on replicationLag to reduce unnecessary warn log